src/OrleansEventSourcing/LogStorage/LogViewAdaptor.cs

```diff
@@ -84,6 +84,17 @@ private void UpdateConfirmedView()
             ConfirmedVersionInternal = GlobalLog.StateAndMetaData.GlobalVersion;
         }
 
+
+        /// <inheritdoc/>
+        public override Task<IReadOnlyList<TLogEntry>> RetrieveLogSegment(int fromVersion, int toVersion)
+        {
+
+            // make a copy of the entries in the range asked for
+            IReadOnlyList<TLogEntry> segment = GlobalLog.StateAndMetaData.Log.GetRange(fromVersion, (toVersion - fromVersion));
```

Comment:
At this point they have already been validated, by the method that calls this one.